### ISSUE TITLE
Slow loading of the product edit page in a shop with a lot of duplicate names in the categories

### ISSUE THREAD

**Issue by yannicka** [PrestaShop/PrestaShop#37639]
### Prerequisites

- [x] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [x] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

I have a shop with many categories (over 40,000) and many duplicate names.

When I go to the admin to edit a product, the system displays the product categories.

To help us discriminate categories with a duplicate name, PrestaShop loads a breadcrumb to display the category name (so it displays "Peugeot > 108" and not just "108").

However, this system is too slow with so many categories with duplicate names.

In my case, the product edit page took over 600 seconds to load.

The issue comes from here:
https://github.com/PrestaShop/PrestaShop/blob/23c159ccd62a4db54a2f93495ce3177e66a2ed1e/src/Core/Category/NameBuilder/CategoryDisplayNameBuilder.php#L77-L98

In my case `$this->getDuplicateNameIds($shopId, $languageId, $useCache);` returns a lot of IDs. And for each ID, it makes multiple queries (more than 65,000 in my case).

### Expected behavior

When there are a lot of duplicates, display "(id) name" and not the category breadcrumbs (I know it's a very very specific issue).

### Steps to reproduce

1. Create a tree of categories (+40,000)
2. Have many categories with the same name
3. Add categories to a product
4. Enjoy slowness

### PrestaShop version(s) where the bug happened

8.1.7

**Comment by paulnoelcholot** [comment]
Hello @yannicka,

I understand your issue :
 ![Image](https://github.com/user-attachments/assets/10cfb529-9351-495d-b43c-9de5a48ea36e)


Did you use the new page product or the older?

Thanks for your feedback

**Comment by yannicka** [comment]
No, it's not (only) in the popup, it's in the edit product page, on the categories block:
![Image](https://github.com/user-attachments/assets/fcf963b6-0267-4eb5-a823-cfb33ff79906)

(I changed the tree to the category ID to fix my problem)

And I the the new page product:
![Image](https://github.com/user-attachments/assets/f3ef5224-8158-4348-8402-1168a1a47925)

**Comment by paulnoelcholot** [comment]
Hello @yannicka 

Thanks for your feedback!

I reproduce the issue with PrestaShop version 8.2.x, I'll add this to the backlog so it can be fixed.

Please be aware that some issues might take a very long time to be resolved. If this one is important to you and you cannot wait for it to be fixed on the project’s own time, we strongly suggest you consider [contacting a professional](https://www.prestashop-project.org/support/) to help you. 

If you fix the issue on your end, please [contribute it back to the project](https://devdocs.prestashop.com/1.7/contribute/contribute-pull-requests/). Remember that the more people contribute, the better PrestaShop becomes for everyone.

Thank you
